### ISSUE TITLE
[Suggestion] Spawn players on the highest valid block rather than the lowest.

### ISSUE THREAD

**Issue by Weedoof** [UnRealDinnerbone/JAMD#11]
As the title.
If players are spawned on the highest valid block rather than the lowest, it would prevent players from spawning inside caves if they're added through data packs, or other mods such as yung's better caves. Personally, I'd appreciate this change since quarries and things are much easier to set up when you don't have to climb ladders to start. And falling down into a cave is easier than climbing a ladder or flying up to the surface should you need to get to where you're not.
A config option for this might be a good idea as well, though I can't deny that it runs the risk of being more effort than its worth, given that it wouldn't make a difference by default anyway.

**Comment by UnRealDinnerbone** [comment]
I will look into to this

**Comment by UnRealDinnerbone** [comment]
Sorry this is taking so long. Im gone this weekend but I WILL take look this next week

**Comment by Weedoof** [comment]
Not a huge thing either way, just something I thought might be nice to have. Thanks for looking into it though :)

**Comment by Nevrai** [comment]
@UnRealDinnerbone Did you ever look at adding this?

@Weedoof Is there a chance you could lend me the datapack for generating caves/dungeons/ores in the mining dim? If you could, that would be much appreciated as I’d love to see an example on how it’s done.

**Comment by Weedoof** [comment]
@DeleteMetaInf yeah, sure. Here's the one I'm using in my modpack. Haven't touched it in over a year though, so don't expect an explanation on what any of it does.

[dimension.zip](https://github.com/UnRealModding/JAMD/files/9791141/dimension.zip)

**Comment by DeleteMetaInf** [comment]
@Weedoof I noticed this is for 1.16.5. So this won’t work in 1.18.2, right?

**Comment by Nevrai** [comment]
@Weedoof I noticed this is for 1.16.5. So this won’t work in 1.18.2, right?

I couldn’t figure out how to get the datapack to work. No ores would spawn in the mining dimension at all. Even the example datapack didn’t work for me (#40).

**Comment by UnRealDinnerbone** [comment]
I plan to port this mod to architectury (forge and fabric) I add this feature then. I should do that this week

**Comment by Nevrai** [comment]
What feature are you talking about? I thought datapacks were supposed to work already.

But if you can update the mod to make it work, that’d be wonderful. Can’t wait! As currently, the mining dimension is just grass, dirt, stone, and bedrock. Couldn’t get datapacks to work.

**Comment by UnRealDinnerbone** [comment]
The spawn hight thing

**Comment by UnRealDinnerbone** [comment]
datpacks should just work but I can double check when updating mod

**Comment by Nevrai** [comment]
I tried the example datapack provided for 1.18.2, but the mining dimension is just empty. There’s only bedrock, deepslate, stone, dirt, and grass.

**Comment by Nevrai** [comment]
After some testing, I figured out what was causing this. Ores do generate in the mining dimension normally, but one mod that I have installed prevents this from happening: [TerraBlender](https://www.curseforge.com/minecraft/mc-mods/terrablender/).

I’ve reported this on that mod’s issue tracker [here](https://github.com/Glitchfiend/TerraBlender/issues/94). It’s an issue with TerraBlender, not JAMD, as it happens with all three mining dimension-adding mods that I’ve tested (JAMD, Advanced Mining Dimension, and Allthemodium). Downgrading to TerraBlender 1.1.0.102 solves the issue and is the only workaround until they fix this.
